### PR TITLE
feat(DTFS2-6718): migrate update underwriters managers decision in tfm to rest

### PR DESCRIPTION
**Pull request by lmarrai-sw**
## Introduction

We want to replace TFM API's GraphQL mutation that updates the underwriter manager's decision with a REST endpoint.

## Resolution

I've created a new endpoint, `PUT /v1/deals/:dealId/underwriting/managers-decision`, in TFM API
This validates that `dealId` is a valid Mongo ID, then executes the existing logic to update the underwriter manager's decision (which updates the deal to add the decision, updates the status of the deal, adds a comment, and conditionally sends an email to PIM, the maker, and the bank).

I've changed TFM UI to use this new endpoint.

### CODE REVIEW

**Review by lmarrai-sw — COMMENTED** [review]
Flagging the TODOs

File: trade-finance-manager-api/src/v1/controllers/deal.controller.update-tfm-underwriter-managers-decision.test.js [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+    api.findBankById = findBankByIdSpy;
+
+    findOneTeamSpy.mockClear();
+    api.findOneTeam = findOneTeamSpy;
+
+    api.updateDeal.mockReset();
+    mockUpdateDeal();
+  });
+
+  afterAll(() => {
```
Comment (AlexBramhill) [inline trade-finance-manager-api/src/v1/controllers/deal.controller.update-tfm-underwriter-managers-decision.test.js:49]: As you mentioned -- probably don't need this

**Comment by AlexBramhill** [comment]
I think we should add very slightly upped error handling -- somewhat in the vein of my previous ticket (maybe less).

What do you think?

File: trade-finance-manager-api/graphql-query-tests/mutation-update-task.api-test.js [hunk 2]
```diff
@@ -57,11 +57,15 @@ describe('graphql mutation - update task', () => {
...
     mockFindUserById();
+
+    api.findOneTeam.mockReset();
+    mockFindOneTeam();
   });
 
   afterAll(() => {
     api.findOneDeal.mockReset();
     api.findUserById.mockReset();
+    api.findOneTeam.mockReset();
```
[the remark this answers is not in the document]
Comment (lmarrai-sw) [reply trade-finance-manager-api/graphql-query-tests/mutation-update-task.api-test.js]: Removed :)

File: trade-finance-manager-api/src/v1/controllers/tasks.api-test.js [hunk 3]
```diff
@@ -48,12 +48,16 @@ describe('tasks controller', () => {
...
+
+    api.findOneTeam.mockReset();
+    mockFindOneTeam();
   });
 
   afterAll(() => {
     api.updateDeal.mockReset();
     api.findOneDeal.mockReset();
     api.findUserById.mockReset();
+    api.findOneTeam.mockReset();
```
[the remark this answers is not in the document]
Comment (lmarrai-sw) [reply trade-finance-manager-api/src/v1/controllers/tasks.api-test.js]: Removed :)

File: trade-finance-manager-api/src/v1/controllers/deal.controller.update-tfm-underwriter-managers-decision.test.js (hunk 1 above)
Comment (lmarrai-sw) [reply trade-finance-manager-api/src/v1/controllers/deal.controller.update-tfm-underwriter-managers-decision.test.js]: Removed, thanks!

File: trade-finance-manager-api/src/v1/controllers/deal-underwriter-managers-decision.controller.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        decision,
+        comments,
+        internalComments,
+        userFullName,
+        timestamp: getTime(new Date()),
+      },
+      stage: decision,
+    },
+  };
+  return api.updateDeal(dealId, managerDecisionUpdate);
```
[the remark this answers is not in the document]
Comment (lmarrai-sw) [reply trade-finance-manager-api/src/v1/controllers/deal-underwriter-managers-decision.controller.js:27]: Yep I'll improve the error handling for this now, thanks!

**Comment by lmarrai-sw** [comment]
Two e2e tests are failing but it's because of a previous PR that was merged into the base branch. We'll fix those e2e tests separately

**Pull request merged**